server/helpers.js

```diff
@@ -1,6 +1,6 @@
-function addProtocolIfMissing(url) {
-  if (url && !url.startsWith('http')) {
-    return `https://${url}`;
+function addProtocolIfMissing(url, protocol = 'https') {
+  if (url && !url.startsWith(protocol)) {
+    return `${protocol}://${url}`;
```

Comment:
if the url starts with `http://`, but the protocal is `https`, this function will end up with adding the protocol again.